app/domain/authentication/authn_gcp/resource_restriction.rb

```diff
@@ -0,0 +1,17 @@
+module Authentication
+  module AuthnGCP
+
+    class ResourceRestriction
```

Comment:
how about implementing this class under the `Authentication` module? you don't need to use it yet in other authenticators (though you can) but it seems that this will be used in other places as well.

Reply:
I think it is part of the effort we should re thinking about post GCP authenticator, because `ExtractResourceRestrictions` also should be in more general module, but right now i want to focus on the GCP and therefore all the logic around the resource restrictions will reside under authGCP module